pkgs/vscode/package.nix

```
@@ -1,10 +1,95 @@
-{ vscode-utils, ... }:
-
-vscode-utils.buildVscodeMarketplaceExtension {
-  mktplcRef = {
-    name = "catppuccin-vsc";
-    publisher = "catppuccin";
-    version = "3.16.1";
-    hash = "sha256-qEwQ583DW17dlJbODN8SNUMbDMCR1gUH4REaFkQT65I=";
+{
+  vscode-utils,
+  sources,
+  nodejs_22,
+  pnpm,
+
+  accentColor ? null,
+  accent ? "mauve",
+  boldKeywords ? true,
+  italicComments ? true,
+  italicKeywords ? true,
+  extraBordersEnabled ? false,
+  workbenchMode ? "default",
+  bracketMode ? "rainbow",
+  colorOverrides ? { },
+  customUIColors ? { },
```

We could also simplify the implementation here and the defaults to the settings submodule above

```suggestion
  catppuccinOptions ? { },
```

Pretty subjective though, so I'll leave it up to you